File: src/main/java/com/yun/kafka/KafkaUtil.java
package com.yun.kafka;

import kafka.admin.AdminUtils;
import kafka.utils.ZKStringSerializer;
import kafka.utils.ZKStringSerializer$;
import kafka.utils.ZkUtils;
import org.I0Itec.zkclient.ZkClient;

import java.util.Properties;

/**
 * Created by caiyunwu on 2017/6/30.
 */
public class KafkaUtil {
    public static void main(String[] args) {
        String zkConnect = "gzns-waimai-do-impala334.gzns.iwm.name:8181";
        int zkSessionTimeout = 60000;
        int zkConnectTimeout = 60000;
        String topic = "create_topic";
//        String topic = "test_topic_create";

        createTopicIfNotExsit(zkConnect, zkSessionTimeout, zkConnectTimeout, topic, 3, 1);


    }

    public static void createTopicIfNotExsit(String zkConnect, int zkSessionTimeout, int zkConnectTimeout, String topic, int partitions, int replicationFactor) {
        ZkClient zkClient = new ZkClient(zkConnect, zkSessionTimeout, zkConnectTimeout, ZKStringSerializer$.MODULE$);
        if (!zkClient.exists(ZkUtils.getTopicPath(topic))) {
            AdminUtils.createTopic(zkClient, topic, partitions, replicationFactor, new Properties());
        } else {
            System.out.println("topic " + topic + " exists");
        }
        zkClient.close();
    }

    public static void createTopicIfNotExsit(String zkConnect, int zkSessionTimeout, int zkConnectTimeout, String topic) {
        createTopicIfNotExsit(zkConnect, zkSessionTimeout, zkConnectTimeout, topic, 20, 1);
    }

    public static void createTopicIfNotExsit(String zkConnect, String topic) {
        createTopicIfNotExsit(zkConnect, 3000, 3000, topic);
    }


    public static void createTopicIfNotExsit(String zkConnect, int zkSessionTimeout, int zkConnectTimeout, String topic, int partitions) {
        createTopicIfNotExsit(zkConnect, zkSessionTimeout, zkConnectTimeout, topic, partitions, 3);
    }


    public static void createTopicIfNotExsit(String zkConnect, String topic, int partitions, int replicationFactor) {
        createTopicIfNotExsit(zkConnect, 3000, 3000, topic, partitions, replicationFactor);
    }

    public static void createTopicIfNotExsit(String zkConnect, String topic, int partitions) {
        createTopicIfNotExsit(zkConnect, 3000, 3000, topic, partitions);
    }
}

File: src/main/scala/kafka/KafkaUtil.scala
package kafka

import java.util
import java.util.Properties

import kafka.consumer.{Consumer, ConsumerConfig}
import kafka.utils.ZkUtils

import scala.collection.mutable
import scala.collection.JavaConverters._

/**
  * Created by caiyunwu on 2017/6/12.
  */
object KafkaUtil {
  def main(args: Array[String]): Unit = {
    val prop=new Properties();
    prop.put("metadata.broker.list","gzns-waimai-do-impala334.gzns.iwm.name:8092")
    prop.put("group.id", "druid-test-sak4")
    prop.put("zookeeper.connect", "gzns-waimai-do-impala334.gzns.iwm.name:8181");
    prop.put("auto.offset.reset", "smallest");
    val topic="druid_waimai_log_fact_log_details_errorlog_v1"

    val conf=new ConsumerConfig(prop)
    val consumerConnector =Consumer.create(conf)
    val topicMap= Map[String,Int]((topic,1))

    val kafkaStreams=consumerConnector.createMessageStreams(topicMap)
    val kafkaStream =  kafkaStreams.get(topic).get(0)
    val iterator= kafkaStream.iterator()

    println("start comsumer")
    var num=0;
    while(iterator.hasNext()) {
      num+=1;
      val  msg= new String(iterator.next().message())
      println(num+":"+msg)
    }



  }
}
